test/fiber/scheduler.rb

```diff
@@ -96,6 +98,16 @@ def run
     @urgent = nil
   end
 
+  def close
+    self.run
+  ensure
+    @closed = true
```

Comment:
I'm thinking it would be good to check this in `#fiber`, so we prohibit creating fibers for this scheduler.
But OTOH since this happens from reassigning Thread#scheduler (or Thread end), `Fiber.schedule {}` won't call this scheduler instance anymore (unless it's used for other Threads, which breaks the whole create/close model).

Reply:
I think the current behaviour + freeze is sufficient.